internal/loader/configmap_loader.go

```diff
@@ -0,0 +1,90 @@
+package loader
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func (l *loader) loadConfigMap(src string, name string, filter string) (string, []string, error) {
+	basePath, err := ioutil.TempDir(l.temporaryDir, name)
+	if err != nil {
+		return "", nil, err
+	}
+
+	filterRegexp, err := regexp.Compile(filter)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "while compiling filter")
+	}
+
+	srcs := strings.Split(src, "/")
+	if len(srcs) != 2 {
+		return "", nil, fmt.Errorf("%s: invalid source format", src)
+	}
+
+	configMap, err := l.getConfigMap(srcs[1], srcs[0])
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "while getting ConfigMap")
+	}
+
+	var fileList []string
+	for key, value := range configMap.Data {
+		if fileList, err = l.copyBytesToFile([]byte(value), key, basePath, filterRegexp, fileList); err != nil {
+			return "", nil, errors.Wrapf(err, "while copying data to file")
+		}
+	}
+
+	for key, value := range configMap.BinaryData {
+		if fileList, err = l.copyBytesToFile(value, key, basePath, filterRegexp, fileList); err != nil {
+			return "", nil, errors.Wrapf(err, "while copying binary data to file")
+		}
+	}
+
+	return basePath, fileList, nil
+}
+
+func (l *loader) getConfigMap(name string, namespace string) (*corev1.ConfigMap, error) {
```

Comment:
Usualy namespace is the first argument then name:
```suggestion
func (l *loader) getConfigMap(namespace, name string) (*corev1.ConfigMap, error) {
```